core/src/transport.cpp

```diff
@@ -129,12 +132,12 @@ namespace irods::http
 		if (_scheme == boost::urls::scheme::http) {
 			return std::make_unique<plain_transport>(_ctx);
 		}
-		else if (_scheme == boost::urls::scheme::https) {
+		if (_scheme == boost::urls::scheme::https) {
 			// There might be a lifetime issue here
 			// TODO: Test too see if it falss apart...
 			auto secure_context{make_secure_context()};
 			return std::make_unique<tls_transport>(_ctx, secure_context);
 		}
-		throw;
+		throw std::invalid_argument{"Scheme is not a supported."};
```

Comment:
How do you trigger this?

Do you just leave off the `http`/`https` in `provider_url`?

Reply:
By changing this:
`<scheme>://...`

We check `url.scheme_id()`, which can be one of the following:
https://www.boost.org/doc/libs/1_84_0/libs/url/doc/html/url/ref/boost__urls__scheme.html

Reply:
Good. Open an issue to add a test for this. Add it to the 0.3.0 milestone.

Reply:
Done.